lib/katello/engine.rb

```diff
@@ -87,15 +87,22 @@ def find_assets(args = {})
       require 'katello/apipie/validators'
     end
 
-    initializer "katello.register_actions", :before => 'foreman_tasks.initialize_dynflow' do |_app|
+    # make sure the Katello plugin is initialized before `after_initialize`
+    # hook so that the resumed Dynflow tasks can rely on everything ready.
+    initializer 'katello.register_plugin', :before => :finisher_hook do
```

Comment:
I don't think the initializers depend on each other here. The only thing that needs to happen is all the initializers before the foreman-tasks' after_initialize hook. The change of ordering was performed mainly because of the faulty calculations of dependencies that Rails did